File: Facade Pattern/1. Minimal/Java/Main.java
public class Main {
    public static void main(String[] args) {
        System.out.println("Facade Pattern in Java");
        System.out.println("----------------------");
        System.out.println();
        
        // Create subsystem classes.
        SubsystemClass1 subsystemClass1 = new SubsystemClass1();
        SubsystemClass2 subsystemClass2 = new SubsystemClass2();

        // Create facade.
        Facade facade = new Facade(subsystemClass1, subsystemClass2);

        // Execute facade operation.
        facade.subsystemOperation();
    }
}

File: Facade Pattern/1. Minimal/Python/main.py
from subsystem_class_1 import SubsystemClass1
from subsystem_class_2 import SubsystemClass2
from facade import Facade

def main():
    print("Facade Pattern in Python")
    print("------------------------")
    print()

    # Create subsystem classes.
    subsystemClass1 = SubsystemClass1()
    subsystemClass2 = SubsystemClass2()

    # Create facade.
    facade = Facade(subsystemClass1, subsystemClass2)

    # Execute facade operation.
    facade.subsystemOperation()

if __name__ == "__main__":
    main()
